src/reputation/distributions.py

```diff
@@ -16,6 +23,90 @@ def amount(self):
     def gives_rep(self):
         return self._give_rep
 
+RSC_YEARLY_GIVEAWAY = 50000000
+MINUTES_IN_YEAR = 525960
+HOURS_IN_YEAR = MINUTES_IN_YEAR / 60
+DAYS_IN_YEAR = 365
+MONTHS_IN_YEAR = 12
+GROWTH = .2
+
+def calculate_upvote_rsc():
+    from paper.models import Vote
+    from discussion.models import (
+        Vote as ReactionVote
+    )
+
+    def calculate_rsc(timeframe, weight):
+        return RSC_YEARLY_GIVEAWAY * weight / timeframe
+
+    def calculate_votes(timeframe):
+        return Vote.objects.filter(vote_type=1, created_date__gte=timeframe).count() + ReactionVote.objects.filter(vote_type=1, created_date__gte=timeframe).count()
+
+    today = datetime.datetime.now(
+        tz=pytz.utc
+    ).replace(
+        hour=0,
+        minute=0,
+        second=0
+    )
+    past_minute = today - datetime.timedelta(minutes=1)
+    past_hour = today - datetime.timedelta(minutes=60)
+    past_day = today - datetime.timedelta(days=1)
+    past_month = today - datetime.timedelta(days=30)
+    past_year = today - datetime.timedelta(days=365)
+
+    votes_in_past_minute = calculate_votes(past_minute)
+    votes_in_past_hour = calculate_votes(past_hour)
+    votes_in_past_day = calculate_votes(past_day)
+    votes_in_past_month = calculate_votes(past_month)
+    votes_in_past_year = calculate_votes(past_year)
+
+    rsc_by_minute = calculate_rsc(votes_in_past_minute * MINUTES_IN_YEAR, .25)
+    rsc_by_hour = calculate_rsc(votes_in_past_hour * HOURS_IN_YEAR, .3)
+    rsc_by_day = calculate_rsc(votes_in_past_day * DAYS_IN_YEAR, .25)
+    rsc_by_month = calculate_rsc(votes_in_past_month * MONTHS_IN_YEAR, .1)
+    rsc_by_year = calculate_rsc(votes_in_past_year, .1)
+
+    rsc_distribute = rsc_by_minute + rsc_by_hour + rsc_by_day + rsc_by_month + rsc_by_year
+    rsc_distribute *= (1 - GROWTH)
+
+    return int(rsc_distribute)
+
+def create_upvote_distribution(vote_type, paper):
+    distribution_amount = calculate_upvote_rsc()
+
+    if paper:
+        from reputation.distributor import Distributor
+        from researchhub_case.models import AuthorClaimCase
+        author_distribution_amount = distribution_amount * .75
+        distribution_amount *= .25 # authors get 75% of the upvote score
+        distributed_amount = 0
+        author_count = paper.true_author_count()
+
+        for author in paper.authors.all():
```

Comment:
Would authors that have not claimed yet only start receiving RSC for upvotes after they claim? or would there be a pot on the side for them to tap into and receive it once claimed?

Reply:
No, we're setting a pot aside for the authors. Upon claiming, they get a share of that pot.